File: cielo24_Java/cielo24/src/test/ActionsTest.java
package test;

import static org.junit.Assert.*;

import java.io.File;
import java.net.URL;
import java.util.ArrayList;
import java.util.logging.Level;
import java.util.logging.Logger;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import cielo24.Actions;
import cielo24.Enums.CaptionFormat;
import cielo24.Enums.Case;
import cielo24.Enums.Fidelity;
import cielo24.Enums.Priority;
import cielo24.json.CreateJobResult;
import cielo24.json.ElementList;
import cielo24.json.ElementListVersion;
import cielo24.json.JobInfo;
import cielo24.json.JobList;
import cielo24.options.CaptionOptions;
import cielo24.utils.Guid;

public class ActionsTest {

	Actions actions = null;
	String username = "api_test";
	String password = "api_test";
	String newPassword = "api_test_new";
	Guid apiToken = null;
	Guid jobId = null;
	Guid taskId = null;
	Guid secureKey = null;
	String sampleVideoUrl = "http://techslides.com/demos/sample-videos/small.mp4";
	String sampleVideoFilePath = "C:\\path\\to\\file.mp4";

	@Test
	public void testOptions() {
		try {
			CaptionOptions co = new CaptionOptions();
			co.captionBySentence = true;
			co.forceCase = Case.upper;
			String[] array = new String[] { "build_url=true", "dfxp_header=header" };
			co.populateFromArray(array);
			assertEquals("build_url=true&caption_by_sentence=true&dfxp_header=header&force_case=upper", co.toQuery());
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Test
	public void testLoginAndLogout() {
		try {
			// Username, password, no headers
			apiToken = actions.login(username, password, false);
			assertEquals(32, apiToken.toString().length());
			// Username, password, headers
			apiToken = actions.login(username, password, true);
			assertEquals(32, apiToken.toString().length());
			// Username, secure key, no headers
			secureKey = actions.generateAPIKey(apiToken, username, false);
			apiToken = actions.login(username, secureKey, false);
			assertEquals(32, apiToken.toString().length());
			// Username, secure key, headers
			apiToken = actions.login(username, secureKey, true);
			assertEquals(32, apiToken.toString().length());
			// Logout
			actions.logout(apiToken);
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
		apiToken = null;
	}

	@Test
	public void testGenerateAndRemoveApiKey() {
		try {
			secureKey = actions.generateAPIKey(apiToken, username, false);
			assertEquals(32, secureKey.toString().length());
			secureKey = actions.generateAPIKey(apiToken, username, true);
			assertEquals(32, secureKey.toString().length());
			actions.removeAPIKey(apiToken, secureKey);
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Test
	public void testUpdatePassword() {
		try {
			actions.updatePassword(apiToken, newPassword);
			actions.updatePassword(apiToken, password);
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Test
	public void testCreateJob() {
		try {
			CreateJobResult result = actions.createJob(apiToken, "testname", "en");
			assertEquals(32, result.jobId.toString().length());
			assertEquals(32, result.taskId.toString().length());
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Test
	public void testAuthorizeJob() {
		try {
			actions.authorizeJob(apiToken, jobId);
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
		jobId = null;
	}

	@Test
	public void testDeleteJob(){
		try {
			taskId = actions.deleteJob(apiToken,jobId);
			assertEquals(32, taskId.toString().length());
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
		jobId = null;
	}

	@Test
	public void testGets(){
		try {
			Logger.getGlobal().log(Level.INFO, "testing getJobInfo()");
			JobInfo info = actions.getJobInfo(apiToken, jobId);
			Logger.getGlobal().log(Level.INFO, "testing getJobList()");
			JobList list = actions.getJobList(apiToken);
			Logger.getGlobal().log(Level.INFO, "testing getElementList()");
			ElementList list2 = actions.getElementList(apiToken, jobId);
			Logger.getGlobal().log(Level.INFO, "testing getListOfElementLists()");
			ArrayList<ElementListVersion> list3 = actions.getListOfElementLists(apiToken, jobId);
			Logger.getGlobal().log(Level.INFO, "testing getMedia()");
			URL url = actions.getMedia(apiToken, jobId);
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Test
	public void testGetText() {
		try {
			actions.getCaption(apiToken, jobId, CaptionFormat.SRT);
			actions.getTranscript(apiToken, jobId);
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Test
	public void testPerformTranscription(){
		try {
			actions.addMediaToJob(apiToken, jobId, new URL(sampleVideoUrl));
			taskId = actions.performTranscription(apiToken, jobId, Fidelity.STANDARD, Priority.STANDARD);
			assertEquals(32, taskId.toString().length());
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Test
	public void testAddData(){
		try {
			File sampleVideoFile = new File(sampleVideoFilePath);
            taskId = actions.addMediaToJob(apiToken, jobId, new URL(sampleVideoUrl));
			assertEquals(32, taskId.toString().length());
			jobId = actions.createJob(apiToken).jobId;
			taskId = actions.addEmbeddedMediaToJob(apiToken, jobId, new URL(sampleVideoUrl));
			assertEquals(32, taskId.toString().length());
			jobId = actions.createJob(apiToken).jobId;
			taskId = actions.addMediaToJob(apiToken,jobId, sampleVideoFile);
			assertEquals(32, taskId.toString().length());
		} catch (Exception e) {
			e.printStackTrace();
			fail("Test failed: " + e.getMessage());
		}
	}

	@Before
	public void setUp() throws Exception {
		this.actions = new Actions("http://sandbox-dev.cielo24.com");
		if (this.apiToken == null){
			this.apiToken = actions.login(username, password, true);
		}
		if (this.jobId == null){
			this.jobId = actions.createJob(apiToken).jobId;
		}
	}

	@After
	public void tearDown() throws Exception {
		// Do nothing
	}
}

File: cielo24_Ruby/cielo24_gem/test/actions_test.rb
require 'test/unit'
require '../lib/cielo24/actions'
require '../lib/cielo24/web_utils'
require '../lib/cielo24/options'
include Cielo24

class AccessTest < Test::Unit::TestCase
  @@actions = Cielo24::Actions.new("http://sandbox-dev.cielo24.com")
  @@password = "testscript2"
  @@username = "testscript"
  @@new_password = "testscript"
  @@api_token = nil
  @@job_id = nil

  # Called before every test method runs. Can be used
  # to set up fixture information.
  def setup
    if @@api_token.nil?
      @@api_token = @@actions.login(@@username, @@password, nil, true)
    end
    if @@job_id.nil?
      @@job_id = @@actions.create_job(@@api_token).JobId
    end
  end

  def test_options
    co = CaptionOptions.new
    co.build_url = true
    co.dfxp_header = "header"
    assert_equal("build_url=true&dfxp_header=header", co.to_query)
  end

  def test_login_and_logout
    # Username, password, no headers
    assert_nothing_raised{@@api_token = @@actions.login(@@username, @@password)}
    assert_equal(32, @@api_token.length)
    # Username, password, headers
    assert_nothing_raised{@@api_token = @@actions.login(@@username, @@password, nil, true)}
    assert_equal(32, @@api_token.length)
    @secure_key = @@actions.generate_api_key(@@api_token, @@username, false)
    # Username, secure key, no headers
    assert_nothing_raised{@@api_token = @@actions.login(@@username, nil, @secure_key)}
    assert_equal(32, @@api_token.length)
    # Username, secure key, headers
    assert_nothing_raised{@@api_token = @@actions.login(@@username, nil, @secure_key, true)}
    assert_equal(32, @@api_token.length)
    # Logout
    assert_nothing_raised{@@actions.logout(@@api_token)}
    #assert_raise(WebError) {@@actions.logout("123")}
    @@api_token = nil
  end

  def test_generate_and_remove_api_key
    assert_nothing_raised{@secure_key = @@actions.generate_api_key(@@api_token, @@username, false)}
    assert_equal(32, @secure_key.length)
    assert_nothing_raised{@secure_key = @@actions.generate_api_key(@@api_token, @@username, true)}
    assert_equal(32, @secure_key.length)
    assert_nothing_raised{@@actions.remove_api_key(@@api_token, @secure_key)}
  end

  def test_update_password
    assert_nothing_raised{@@actions.update_password(@@api_token, @@new_password)}
    assert_nothing_raised{@@actions.update_password(@@api_token, @@password)}
    assert_raise(WebError) {@@actions.update_password("not_an_api_token", @@password)}
  end

  def test_create_job
    assert_nothing_raised{@mash = @@actions.create_job(@@api_token,"testname","en")}
    assert_equal(32, @mash.JobId.length)
    assert_equal(32, @mash.TaskId.length)
  end

  def test_authorize_job
    assert_nothing_raised{@@actions.authorize_job(@@api_token,@@job_id)}
    @@job_id = nil
  end

  def test_delete_job
    assert_nothing_raised{@task_id = @@actions.delete_job(@@api_token,@@job_id)}
    assert_equal(32, @task_id.length)
    @@job_id = nil
  end

  def test_gets
    assert_nothing_raised{@mash = @@actions.get_job_info(@@api_token,@@job_id)}
    assert_nothing_raised{@@actions.get_job_list(@@api_token)}
    assert_nothing_raised{@@actions.get_element_list(@@api_token, @@job_id)}
    assert_nothing_raised{@@actions.get_list_of_element_lists(@@api_token, @@job_id)}
    assert_nothing_raised{@@actions.get_media(@@api_token, @@job_id)}
  end

  def test_get_text
    assert_nothing_raised{@@actions.get_caption(@@api_token,@@job_id, "SRT")}
    assert_nothing_raised{@@actions.get_transcript(@@api_token,@@job_id)}
  end

  def test_perform_transcription
    assert_nothing_raised{@task_id = @@actions.add_media_to_job_url(@@api_token,@@job_id,"http://lesmoralesphotography.com/cielo24/test_suite/End_to_End_Regression/media_short_2327da9786d44a9a9c62242853593059.mp4")}
    assert_equal(32, @task_id.length)
    assert_nothing_raised{@task_id = @@actions.perform_transcription(@@api_token,@@job_id, "PREMIUM", "STANDARD")}
    assert_equal(32, @task_id.length)
  end

  def test_add_data
    assert_nothing_raised{@task_id = @@actions.add_media_to_job_url(@@api_token,@@job_id,"http://lesmoralesphotography.com/cielo24/test_suite/End_to_End_Regression/media_short_2327da9786d44a9a9c62242853593059.mp4")}
    assert_equal(32, @task_id.length)
    @@job_id = @@actions.create_job(@@api_token).JobId
    assert_nothing_raised{@task_id = @@actions.add_media_to_job_embedded(@@api_token,@@job_id,"http://lesmoralesphotography.com/cielo24/test_suite/End_to_End_Regression/media_short_2327da9786d44a9a9c62242853593059.mp4")}
    assert_equal(32, @task_id.length)
    @@job_id = @@actions.create_job(@@api_token).JobId
    file = File.open("C:/Users/Evgeny/Videos/The_Hobbit_480p.mov", "r")
    #file = File.open("C:/Users/Evgeny/Videos/Thor.The.Dark.World.2013.1080p.BluRay.x264.YIFY.mp4", "r")
    #file = File.open("C:/Users/Evgeny/Videos/small.mp4", "r")
    assert_nothing_raised{@task_id = @@actions.add_media_to_job_file(@@api_token,@@job_id, file)}
    assert_equal(32, @task_id.length)
  end

  # Called after every test method runs. Can be used to tear down fixture information.
  def teardown
    # Do nothing
  end
end